civis/io/_files.py

```diff
@@ -160,25 +160,28 @@ def _upload_part(item):
 
     # upload each part
     try:
-        with tempfile.TemporaryDirectory() as tmp_dir:
+        pool = Pool(MAX_THREADS)
+        with TemporaryDirectory() as tmp_dir:
+            tmp_path = os.path.join(tmp_dir, 'file_to_civis_{}.csv')
             for i in range(0, num_parts):
                 offset = part_size * i
                 num_bytes = min(part_size, file_size - offset)
                 buf.seek(offset)
 
                 # write part to disk so that we can stream it
-                file_in = os.path.join(tmp_dir, str(i) + '.csv')
+                file_in = tmp_path.format(i)
                 with open(file_in, 'wb') as fin:
                     for x in _gen_chunks(buf, num_bytes):
                         fin.write(x)
 
-            with Pool(MAX_THREADS) as pool:
-                pool.map(_upload_part, enumerate(urls))
+            _upload_part = partial(_upload_part_base, part_path=tmp_path)
+            pool.map(_upload_part, enumerate(urls))
 
     # complete the multipart upload; an abort will be triggered
     # if any part except the last failed to upload at least 5MB
     finally:
         client.files.post_multipart_complete(file_response.id)
+        pool.terminate()
```

Comment:
Could you put the `pool.terminate()` as the first line of the `finally`? It seems less likely to error than the client call. Alternately, the pool creation and destruction could be in its own small try/finally block where the context manager used to be.

Reply:
+1

moved